modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+xref:db/db-connector-ddl.adoc[Execute DDL Statements].
+
+Given that the table is already created and that a there is a
+xref:db/db-configure-connection.adoc[connection configuration] named `dbConfig`,
+the following operation inserts a new record in the *electronic* table:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig">
+  <db:sql>
+    INSERT INTO electronic(name, description, price, discount)
+    VALUES ('Coffee Machine', 'Model: XYZ99. Uses small size capsules.', 120, 5)
+  </db:sql>
+</db:insert>
+----
+
+Note that the *id* parameter is not passed in the query because it is an
+autoincrement key that is generated automatically by the database.
+
+[NOTE]
+=============
+To benefit from the advantages of
+xref:db/db-connector-query.adoc#_use_input_parameters_to_protect_database_queries[input parameters],
+structure the equivalent query as follows:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig">
+    <db:sql>
+        INSERT INTO electronic(name, description, price, discount)
+        VALUES (:name, :description, :price, :discount)
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+=============
+
+The insert operation outputs a `Statement Result` object that contains two fields: an _affectedRows_
+integer that indicates how many rows were affected by the query, and a _generatedKeys_ map
+that contains the autogenerated keys. Given this example, the output is as follows:
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+The _generatedKeys_ map is empty, although a row is successfully added to the table.
+The map is empty because this information is disabled by default to avoid overhead.
+To activate the information and generate keys, add the `autoGenerateKeys` parameter, and then set it to true:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true">
+   <db:sql>
+       INSERT INTO electronic(name, description, price, discount)
+       VALUES (:name, :description, :price, :discount)
+   </db:sql>
+   <db:input-parameters>#[{
+       name: 'Coffee Machine',
+       description: 'Model:XYZ99. Uses small size capsules.',
+       price: 120,
+       discount: 5}]
+   </db:input-parameters>
+</db:insert>
+----
+
+If the statement generates multiple keys per affected row, you can specify which columns
+should be returned by setting the parameter `autoGeneratedKeysColumnNames` to a list that contains
+the desired column names. For example, to ensure that only the _id_ column is
+returned, set the parameter to a DataWeave list with a single string element:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true" autoGeneratedKeysColumnNames="#[['id']]">
+    ...
+</db:insert>
+----
+
+== Update Example
```

Comment:
Again, retitle without the word, Example.

Reply:
Removed.